clouddriver-core/src/main/java/com/netflix/spinnaker/clouddriver/security/AccountDefinitionSecretManager.java

```diff
@@ -16,34 +16,72 @@
 
 @RequiredArgsConstructor
+@NonnullByDefault
 public class AccountDefinitionSecretManager {
-  private final SecretManager secretManager;
-  private final AccountDefinitionAuthorizer authorizer;
+  @Delegate private final UserSecretManager userSecretManager;
+  private final AccountSecurityPolicy policy;
 
-  public String getEncryptedSecret(String encryptedUri) {
-    return EncryptedSecret.isEncryptedFile(encryptedUri)
-        ? secretManager.decryptAsFile(encryptedUri).toString()
-        : secretManager.decrypt(encryptedUri);
+  private final Map<String, Set<UserSecretReference>> refsByAccountName = new ConcurrentHashMap<>();
+
+  /**
+   * Gets a user secret string value for the given account. User secret references are tracked
+   * through this method to support time-of-use access control checks for accounts after they've
+   * been loaded by the system.
+   *
+   * @param reference parsed user secret reference to decrypt
+   * @param accountName name of account requesting the user secret
+   * @return the contents of the requested user secret string
+   */
+  public String getUserSecretString(UserSecretReference reference, String accountName) {
+    var secret = getUserSecret(reference);
+    refsByAccountName
+        .computeIfAbsent(accountName, ignored -> ConcurrentHashMap.newKeySet())
+        .add(reference);
+    var parameterName = StandardSecretParameter.KEY.getParameterName();
+    var secretKey = reference.getParameters().getOrDefault(parameterName, "");
+    try {
+      return secret.getSecretString(secretKey);
+    } catch (NoSuchElementException e) {
+      throw new SecretDecryptionException(e);
+    }
   }
 
-  public boolean canAccessAccountWithSecrets(@Nonnull String accountName) {
-    var username = SecurityContextHolder.getContext().getAuthentication().getName();
-    if (authorizer.isAdmin(username)) {
+  /**
+   * Indicates if the given username is authorized to access the given account. When Fiat is
+   * enabled, this allows admins to access accounts along with users who have both WRITE permission
+   * on the account and are authorized to use any provided {@link UserSecretReference} data.
+   *
+   * @param username username to check for authorization to use the given account
+   * @param accountName the name of the account to check access to
+   * @return true if the given username is allowed to access the given account
+   */
+  public boolean canAccessAccountWithSecrets(String username, String accountName) {
+    if (policy.isAdmin(username)) {
       return true;
     }
-    var userRoles = authorizer.getRoles(username);
+    var userRoles = policy.getRoles(username);
     if (userRoles.isEmpty()) {
       return false;
     }
-    // TODO(jvz): update with https://github.com/spinnaker/kork/pull/942
-    //  to add user secrets usage tracking for time of use authz checks
-    return authorizer.canAccessAccount(username, accountName);
+    if (refsByAccountName.getOrDefault(accountName, Set.of()).stream()
```

Comment:
I could use some comments here...mostly a reminder that Collections.disjoint returns true if there are no common elements...meaning that none of the userRoles provide access to the secret.